invisible_cities/cities/detsim_waveforms.py

```diff
@@ -0,0 +1,74 @@
+import numpy as np
+from warnings import warn
+
+##################################
+######### WAVEFORMS ##############
+##################################
+def create_waveform(times    : np.ndarray,
+                    pes      : np.ndarray,
+                    bins     : np.ndarray,
+                    nsamples : int) -> np.ndarray:
+    """
+    This function builds a waveform from a set of (buffer_time, pes) values.
+    This set is of values come from the times and pes arguments.
+
+    Input
+    times: a vector with the buffer times in which a photoelectron is produced in the
+    detector.
+    pes (from photoelecton): a vector with the photoelectrons produced in the detector in
+    each of the buffer times in times argument.
+    bins: a vector with the output waveform bin times (for example [0, 25, 50, ...] if
+    the detector has a sampling time of 25).
+    nsamples: an integer that controlls the distribution of the photoelectrons in each of
+    the waveform bins. The counts (N) in a given time bin (T) are distributed in the waveform
+    such as the nsamples posterior to T would have N/nsamples counts (included T).
+    nsamples must be >=1 an <len(bins).
+    """
+    if nsamples<1:
+        raise Exception("nsamples<1. nsamples must be >= 1")
+    if nsamples>len(bins):
+        warn("nsamples > number of bins. Shrinking it to number of bins")
+        nsamples = len(bins)
+
+    wf = np.zeros(len(bins))
+    if np.sum(pes)==0:
+        return wf
+
+    t = np.repeat(times, pes)
+    sel = (bins[0]<=t) & (t<=bins[-1])
```

Comment:
```suggestion
    sel = in_range(t, bins[0], bins[-1])
```
`in_range` from core_functions

Reply:
It would be `sel = in_range(t, bins[0], bins[-1], right_closed=True)`. Nevertheless, I do not see a strong reason for this line to be changed, since in my opinion it is clearer for a code reader/reviewer to read as it is than looking to `in_range` to figure out what this function does.

Reply:
> It would be sel = in_range(t, bins[0], bins[-1], right_closed=True).

Not if you use bin edges.

> Nevertheless, I do not see a strong reason for this line to be changed, since in my opinion it is clearer for a code reader/reviewer to read as it is than looking to in_range to figure out what this function does.

This function has been part of the IC utilities for very long. It was included because that functionality is required in many cases.Therefore we have a somewhat tricky piece of code that
- is not repeated everywhere (typos less likely)
- is tested
- has simpler syntax

It is ok if you didn't know about this function, but it has some clear benefits even if the first time you have to invest some time in learning it. Maybe the name should be changed to `in_interval`, which would be more appropriate.